01_fib/starfallz/main.go

```diff
@@ -0,0 +1,44 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"strings"
+)
+
+var out io.Writer = os.Stdout
+
+func main() {
+	fmt.Fprint(out, strings.Trim(strings.Join(strings.Split(fmt.Sprint(fib(7)), " "), "\n"), "[]"))
```

Comment:
That's a novel approach :)

However, the lab requirement was to write a function with the following signature:  
`func fib(n int)` 
happy to let this go.

You could pull the `Split` and `Join` together with a `strings.ReplaceAll`.
I agree it's nice that you don't print inside `fib`, however, I'd wrap the printing in a `printIntSlice` function or similar, that iterates through the `fib`'s result array `Println`s each element. 
In order to understand your approach the reader of your code has to have intimate knowledge with the way golang prints slices.

Reply:
Yeah initially I did that however it felt kind of wierd to do many things in one function. I will implement the pritntIntSlice func instead since I want to be able to test them separately and not do too much in one func

Reply:
Sounds good!